### DISCUSSION TITLE
Custom local-only judgements during gameplay and after watching a replay

### DISCUSSION THREAD

**Discussion by Newbie1st** [ppy/osu#31020] [Ideas]
My all-time favourite feature in rhythm games is Quaver's custom judgements system, which allows the player to use custom values for hit windows shown during gameplay and at the results screen. It's completely different from the current Difficulty Adjust mod, due to it not affecting actual scoredata, as all of the hits are saved, calculated and sent to the server in the default judgement and only locally converted to the custom one.

![1292024 9-45-44-128](https://github.com/user-attachments/assets/7483d969-9f2e-4b42-91fb-b8ad3147568b)
Notice how the performance rating and ranked accuracy values remain the same after conversion
![1292024 9-46-15-912](https://github.com/user-attachments/assets/e0b022ec-afc5-45f7-8057-ce6af00f7da0)

Using a harsher custom judgement gives the player better feedback during gameplay and helps them train accuracy beyond OD10 (which is especially useful for mania players, considering that MAX hit window remains the same no matter the OD). Another way it can be used is seeing how the score would look like if played in a different game, Stepmania Judge 7 on the screenshot above, for example, is also known as Ridiculous judgement, it essentially adds another judgement above Marvelous with the hit window half the size, while shifing all the others down a step, compared to the default Judge 4 (Marvelous becomes Perfect, Perfect - Good).

What custom judgement editor looks like in Quaver
![1292024 10-13-4-400](https://github.com/user-attachments/assets/3812a05e-29e0-4e6e-94b4-d2d62f891dcc)

This would additionally require adding support for separate visual-only hit result calculation, preventing players from swapping the scoredata sent to the server to the one played in more lenient judgement and making osu save the hit distribution after watching a replay (might already be in the game).

The "convert to" dropdown menu can be put at the cursor position in the screenshot
![osu_2024-12-09_10-25-32](https://github.com/user-attachments/assets/2e528f74-1d92-49cb-b489-3a0c989a9c23)

I really miss this feature, considering that Quaver can't simply fetch my lazer beatmaps, like it could (and still can) with stable, playing std maps with low OD feels very sloppy and easier to shitmiss due to the lack of feedback and I'm also tired of being forced to manually convert unranked mania maps to OD9, since increasing the OD through DA still applies 0.5x multiplier, which messes up my local leaderboards.

**Comment by hellidox** [comment]
Please god
